src/DatePicker/Utilities.elm

```diff
@@ -314,3 +317,69 @@ toUtcDateTimeString datetime =
         ++ ":"
         ++ addLeadingZero (Time.toSecond Time.utc datetime)
         ++ " (UTC)"
+
+
+setDayNotTime : Maybe Posix -> Posix -> Posix
+setDayNotTime prevPickedDT newPickedDT =
+    let
+        ( prevPickedHour, prevPickedMinute ) =
+            Maybe.map (Time.posixToParts Time.utc) prevPickedDT
+                |> Maybe.map (\parts -> ( parts.hour, parts.minute ))
+                |> Maybe.withDefault ( 0, 0 )
+    in
+    Time.posixToParts Time.utc newPickedDT |> (\parts -> { parts | hour = prevPickedHour, minute = prevPickedMinute }) |> Time.partsToPosix Time.utc
+
+
+type HourOrMinute
+    = IsHour Int
+    | IsMinute Int
+
+
+setTimeNotDay : Posix -> HourOrMinute -> Posix
+setTimeNotDay timeToUpdate hourOrMinute =
```

Comment:
You can make this simpler by just creating two functions. Then you don't need the HourOrMinute type and the functions both become much simpler.